pkgs/desktops/cinnamon/cjs/default.nix

```diff
@@ -0,0 +1,83 @@
+{ autoconf-archive
+, autoreconfHook
+, dbus-glib
+, fetchFromGitHub
+, gobject-introspection
+, pkgconfig
+, stdenv
+, wrapGAppsHook
+, python3
+, cairo
+, gnome3
+, xapps
+, keybinder3
+, upower
+, callPackage
+, glib
+, libffi
+, gtk3
+}:
+
+let
+
+  # https://github.com/linuxmint/cjs/issues/80
+  spidermonkey_52 = callPackage ./spidermonkey_52.nix {};
+
+in
+
+stdenv.mkDerivation rec {
+  pname = "cjs";
+  version = "4.4.0";
+
+  src = fetchFromGitHub {
+    owner = "linuxmint";
+    repo = pname;
+    rev = version;
+    sha256 = "0q5h2pbwysc6hwq5js3lwi6zn7i5qjjy070ynfhfn3z69lw5iz2d";
+  };
+
+  propagatedBuildInputs = [
+    glib
+  ];
+
+  nativeBuildInputs = [
+    autoconf-archive
+    autoreconfHook
+    wrapGAppsHook
+    pkgconfig
+  ];
+
+  buildInputs = [
+    # from .pc
+    gobject-introspection
+    libffi
+    # gthread-2.0 ?
+    # gio-2.0 ?
+    spidermonkey_52 # mozjs-52
+    cairo # +cairo-gobject
+    gtk3
+
+    # other
+
+    dbus-glib
+
+    # bindings
+    # (python3.withPackages (pp: with pp; [ setproctitle pygobject3 pycairo ]))
+    gnome3.caribou
+    keybinder3
+    upower
+    xapps
```

Comment:
Where are these needed in CJS? I'm thinking maybe it's their typelib's in `GI_TYPELIB_PATH` are what's needed.

Reply:
Yes, I suppose that's it

Reply:
Should I keep them? Is there a way to just add them into the typelib path?

Reply:
What error does it produce when omitted?

Reply:
When a .js file does things like `require("xapps")` it complains that it's typelib can't be found

Though I don't know if the typelibs should be added here or wrapped/added via env in another package

Reply:
If the error occurs on importing xapps, the packages should be added to `propagateBuildInputs` of `xapps` and `wrapGAppsHook` should be added to the application importing xapps.

Reply:
Not just xapps, all the others as well

Reply:
Well every runtime dependency that is not resolved at build time (e.g. hardcoded by a patch or a linker) needs to be propagated to be handled by a wrapper.